### PR TITLE
Configure git to use ssh rather than unsupported `git://` protocol

### PR DESCRIPTION
**Pull request by mickmister**
#### Summary

As an alternative to https://github.com/mattermost/circleci-orbs/pull/30, this PR makes it so git is configured in a way that sidesteps this issue https://github.com/npm/cli/issues/4896.

### CODE REVIEW

**Comment by mickmister** [comment]
@hanzei Please a look at this as well. A few months ago, plugin CI got much slower due to fetching npm dependencies from GitHub being very slow. The fix in this PR changes the protocol to sidestep the issue https://github.com/npm/cli/issues/4896. With this fix in the Jira plugin, it made CI go from 34 minutes to 8 minutes.

**Comment by mickmister** [comment]
@hanzei I'd like to get this and https://github.com/mattermost/circleci-orbs/pull/32 merged to fix a lot of the CI-related issues with plugins. Plugins will now be able to be compiled with Go 1.18, and will work with any node version.

**Review by hanzei — APPROVED** [review]
LGTM

File: plugin-ci/orb.yml [hunk 1]
```diff
@@ -71,6 +71,7 @@ jobs:
     resource_class: xlarge
     steps:
       - checkout
+      - run: git config --global url."ssh://git@".insteadOf git://
```
Comment (hanzei) [inline plugin-ci/orb.yml]: Maybe this should be a command instead so the code is more DRY.

**Review by hanzei — APPROVED** [review]

**Pull request merged**